### PR TITLE
Feature/dapp 432

### PR DESCRIPTION
**Pull request by leodhb**
# Description
- We need to set the donation cooldown dynamically

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] It is urgent
- [ ] <span style="color:#d44037">Dangerous change</span>

### CODE REVIEW

File: app/commands/donations/donate.rb [hunk 1]
```diff
@@ -15,19 +15,31 @@ def initialize(integration:, non_profit:, user:)
...
       with_exception_handle do
-        Donation.transaction do
-          create_donation
-          @transaction_hash = create_blockchain_donation
-          set_user_last_donation_at
-          create_donation_blockchain_transaction(transaction_hash)
+        if allowed?
+          transact_donation
+        else
+          errors.add(:message, 'The current user cannot donate')
```
Comment (NickNish09) [inline app/commands/donations/donate.rb:21]: maybe we could use the I18n message here `I18n.t('donations.blocked_message')`. What do you think

File: spec/models/user_donation_stats_spec.rb [hunk 2]
```diff
@@ -62,7 +82,23 @@
       end
 
       it 'returns true' do
```
Comment (NickNish09) [inline spec/models/user_donation_stats_spec.rb:84]: I think this comes from past PRs, but if we can change here to 'returns false' kk

File: spec/models/user_donation_stats_spec.rb [hunk 3]
```diff
@@ -62,7 +82,23 @@
...
+
+      let(:user_donation_stats) do
+        build(:user_donation_stats, last_donation_at: DateTime.parse('2021-01-13 10:00:00'))
+      end
+
+      before do
+        allow(Time.zone).to receive(:now).and_return(DateTime.parse('2021-01-13 10:40:00'))
+      end
+
+      it 'returns true' do
```
Comment (NickNish09) [inline spec/models/user_donation_stats_spec.rb:100]: ditto

File: app/commands/donations/donate.rb (hunk 1 above)
Comment (leodhb) [reply app/commands/donations/donate.rb]: Done!

File: spec/models/user_donation_stats_spec.rb (hunk 2 above)
Comment (leodhb) [reply spec/models/user_donation_stats_spec.rb]: Done!

File: spec/models/user_donation_stats_spec.rb (hunk 3 above)
Comment (leodhb) [reply spec/models/user_donation_stats_spec.rb]: Done!

**Review by NickNish09 — APPROVED** [review]

**Pull request merged**